lib/pages/completed_tasks.dart

```diff
@@ -6,90 +6,108 @@ import 'package:task_list_app/constants/constants.dart';
+          backgroundColor: bgColor,
+          appBar: AppBar(
+            backgroundColor: Colors.black,
+            title: Text(
+              'Completed Tasks',
+              style: TextStyle(color: white),
+            ),
+            centerTitle: true,
           ),
-          centerTitle: true,
-        ),
-        body: ListView.builder(
-          itemCount: completedTasks.length,
-          itemBuilder: (context, index) {
-            return completedTasks[index];
-          },
-        ),
-      );
-    });
+          body: ListView.builder(
+            itemCount: completedTasks.length,
+            itemBuilder: (context, index) {
+              return completedTasks[index];
+            },
+          ),
+        );
+      },
+    );
   }
 
-  void _onTapTrashIcon(Task task) {
+  void _onTapTrashIcon(BuildContext context, Task task) {
     showDialog(
       context: context,
       builder: (context) => AlertDialog(
-          surfaceTintColor: bgColor,
-          backgroundColor: bgColor,
-          title: Text(
-            'Mark as Incomplete or Delete Task?',
-            style: TextStyle(color: white),
-          ),
-          actions: [
-            TextButton(
-              child: Text(
-                'Mark Incomplete',
-                style: TextStyle(color: white),
-              ),
-              onPressed: () => context.pop(),
+        key: const Key('incomplete_or_delete'),
+        surfaceTintColor: bgColor,
+        backgroundColor: bgColor,
+        title: Text(
+          'Mark as Incomplete or Delete Task?',
+          style: TextStyle(color: white),
+        ),
+        actions: [
+          TextButton(
+            key: const Key('complete_mark_incomplete'),
+            child: Text(
+              'Mark Incomplete',
+              style: TextStyle(color: white),
             ),
-            TextButton(
-              child: Text(
-                'Delete Permanently',
-                style: TextStyle(color: white),
-              ),
-              onPressed: () => context.pop(),
+            onPressed: () async {
+              context
+                  .read<TaskCubit>()
+                  .updateTask(task: task, newStatus: TaskStatus.todo)
+                  .then((result) {
+                context
+                    .read<TaskCubit>()
+                    .readTasksByStatus(TaskStatus.completed);
+                context.pop();
+              });
+            },
+          ),
+          TextButton(
+            key: const Key('complete_mark_deleted'),
+            child: Text(
+              'Delete Permanently',
+              style: TextStyle(color: white),
             ),
-          ]),
+            onPressed: () async {
+              context.read<TaskCubit>().deleteTask(task: task).then((result) {
+                context
+                    .read<TaskCubit>()
+                    .readTasksByStatus(TaskStatus.completed);
+                context.pop();
+              });
+            },
```

Comment:
Consider putting this handler in a separate, private function.

Reply:
done